bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java

```diff
@@ -152,11 +154,36 @@ public static BError createError(Throwable error) {
      * @throws BError if given error type is not defined in the ballerina module.
      */
     public static BError createError(Module module, String errorTypeName,
-                                     BString message, BError cause, Object details) {
+                                     BString message, BError cause, BMap<BString, Object> details) {
         ValueCreator valueCreator = ValueCreator.getValueCreator(ValueCreator.getLookupKey(module));
         return valueCreator.createErrorValue(errorTypeName, message, cause, details);
     }
 
+    /**
+     * Create an error value with given error type defined in the given module and reason, cause and details.
+     *
+     * @param module        module name
+     * @param errorTypeName error type name
+     * @param message       error message
+     * @param cause         error cause
+     * @param details       error details
+     * @return error value
+     */
+    @Deprecated
+    public static BError createError(Module module, String errorTypeName,
+                                     BString message, BError cause, BString details) {
+        ValueCreator valueCreator = ValueCreator.getValueCreator(ValueCreator.getLookupKey(module));
+        MappingInitialValueEntry[] initialValues;
+        if (details != null) {
+            initialValues = new MappingInitialValueEntry[1];
+            initialValues[0] = new MappingInitialValueEntry.KeyValueEntry(ERROR_MESSAGE_FIELD, details);
+        } else {
+            initialValues = new MappingInitialValueEntry[0];
+        }
+        MapValueImpl<BString, Object> detailMap = new MapValueImpl(PredefinedTypes.TYPE_ERROR_DETAIL, initialValues);
+        return valueCreator.createErrorValue(errorTypeName, message, cause, detailMap);
```

Comment:
Shall we call createError(errorTypeName, message, cause, detailMap)

Reply:
Addressed with https://github.com/gabilang/ballerina-lang/commit/ef50df79f7000f89a78caa3c2152af971c366e29